tap_bing_ads/__init__.py

```diff
@@ -54,6 +58,40 @@
 
 ARRAY_TYPE_REGEX = r'ArrayOf([A-Za-z0-9]+)'
 
+def should_retry_httperror(exception):
+    """ Return true if exception is required to retry otherwise return false """
+    try:
+        if isinstance(exception, ConnectionError) or isinstance(exception, ssl.SSLError):
+            return True
+        elif isinstance(exception, suds.transport.TransportError) or isinstance(exception, socket.timeout):
+            return True
+        elif type(exception) == URLError:
+            return True
+        elif (type(exception) == Exception and exception.args[0][0] == 408) or exception.code == 408:
+            # A 408 Request Timeout is an HTTP response status code that indicates the server didn't receive a complete
+            # request message within the server's allotted timeout period. A suds SDK catches HTTPError with status code 408 and
+            # raises Exception: (408, 'Request Timeout'). That's why we retrying this error also.
+            return True
+        return 500 <= exception.code < 600
+    except AttributeError:
+        return False
+
+def bing_ads_error_handling(fnc):
+    """
+        Retry following errors for 60 seconds,
+        socket.timeout, ConnectionError, internal server error(500-range), SSLError, URLError, HTTPError(408), Transport errors.
+        Raise the error directly for all errors except mentioned above errors.
+    """
+    @backoff.on_exception(backoff.expo,
+                          (Exception),
+                          giveup=lambda e: not should_retry_httperror(e),
+                          max_time=60, # 60 seconds
+                          factor=2)
```

Comment:
@prijendev How many times will the exception retry because of this backoff?
Can you please add comment above this exception regarding the behavior of backoff mechanism

Reply:
Here, we have implemented backoff based on `max_time`. It retries errors until 60 seconds(7 or 8 times retry perform). If after passing 60 seconds the same error comes, then it will be raised. Added comment in the code also.